builtin/logical/pki/path_acme_test.go

```diff
@@ -322,76 +322,110 @@ func TestAcmeBasicWorkflowWithEab(t *testing.T) {
-
-	infoForKid := keyInfo[kid].(map[string]interface{})
-	keyBits := infoForKid["key_bits"].(json.Number)
-	require.Equal(t, "256", keyBits.String())
-	require.Equal(t, "hs", infoForKid["key_type"])
-
-	// Create new account with EAB
-	t.Logf("Testing register on %s", baseAcmeURL)
-	_, err = acmeClient.Register(testCtx, acct, func(tosURL string) bool { return true })
-	require.NoError(t, err, "failed registering new account with eab")
+			// Create new account without EAB, should fail
+			t.Logf("Testing register on %s", baseAcmeURL)
+			_, err = acmeClient.Register(testCtx, &acme.Account{}, func(tosURL string) bool { return true })
+			require.ErrorContains(t, err, "urn:ietf:params:acme:error:externalAccountRequired",
+				"expected failure creating an account without eab")
+
+			// Test fetch, list, delete workflow
+			kid, _ := getEABKey(t, client, tc.prefixUrl)
+			resp, err := client.Logical().ListWithContext(testCtx, "pki/eab")
+			require.NoError(t, err, "failed to list eab tokens")
+			require.NotNil(t, resp, "list response for eab tokens should not be nil")
+			require.Contains(t, resp.Data, "keys")
+			require.Contains(t, resp.Data, "key_info")
+			require.Len(t, resp.Data["keys"], 1)
+			require.Contains(t, resp.Data["keys"], kid)
+
+			_, err = client.Logical().DeleteWithContext(testCtx, "pki/eab/"+kid)
+			require.NoError(t, err, "failed to delete eab")
+
+			// List eabs should return zero results
+			resp, err = client.Logical().ListWithContext(testCtx, "pki/eab")
+			require.NoError(t, err, "failed to list eab tokens")
+			require.Nil(t, resp, "list response for eab tokens should have been nil")
+
+			// fetch a new EAB
+			kid, eabKeyBytes := getEABKey(t, client, tc.prefixUrl)
+			acct := &acme.Account{
+				ExternalAccountBinding: &acme.ExternalAccountBinding{
+					KID: kid,
+					Key: eabKeyBytes,
+				},
+			}
 
-	// Make sure our EAB is no longer available
-	resp, err = client.Logical().ListWithContext(context.Background(), "pki/acme/eab")
-	require.NoError(t, err, "failed to list eab tokens")
-	require.Nil(t, resp, "list response for eab tokens should have been nil due to empty list")
+			// Make sure we can list our key
+			resp, err = client.Logical().ListWithContext(testCtx, "pki/eab")
+			require.NoError(t, err, "failed to list eab tokens")
+			require.NotNil(t, resp, "list response for eab tokens should not be nil")
+			require.Contains(t, resp.Data, "keys")
+			require.Contains(t, resp.Data, "key_info")
+			require.Len(t, resp.Data["keys"], 1)
+			require.Contains(t, resp.Data["keys"], kid)
+
+			keyInfo := resp.Data["key_info"].(map[string]interface{})
+			require.Contains(t, keyInfo, kid)
+
+			infoForKid := keyInfo[kid].(map[string]interface{})
+			keyBits := infoForKid["key_bits"].(json.Number)
+			require.Equal(t, "256", keyBits.String())
+			require.Equal(t, "hs", infoForKid["key_type"])
+			require.Equal(t, tc.prefixUrl, infoForKid["acme_directory"])
+
+			// Create new account with EAB
```

Comment:
Want to make sure we can't register an account with a good EAB for a different directory?